mesonbuild/backend/backends.py

```diff
@@ -467,18 +467,21 @@ def relpath(todir: str, fromdir: str) -> str:
 
     def flatten_object_list(self, target: build.BuildTarget, proj_dir_to_build_root: str = ''
                             ) -> T.Tuple[T.List[str], T.List[build.BuildTargetTypes]]:
-        obj_list, deps = self._flatten_object_list(target, target.get_objects(), proj_dir_to_build_root)
+        obj_list: T.List[str] = []
+        deps: T.List[build.BuildTargetTypes] = []
+        self._flatten_object_list(target, target.get_objects(), proj_dir_to_build_root, obj_list, deps)
         return list(dict.fromkeys(obj_list)), deps
 
     def determine_ext_objs(self, objects: build.ExtractedObjects, proj_dir_to_build_root: str = '') -> T.List[str]:
-        obj_list, _ = self._flatten_object_list(objects.target, [objects], proj_dir_to_build_root)
+        obj_list: T.List[str] = []
+        deps: T.List[build.BuildTargetTypes] = []
+        self._flatten_object_list(objects.target, [objects], proj_dir_to_build_root, obj_list, deps)
         return list(dict.fromkeys(obj_list))
 
     def _flatten_object_list(self, target: build.BuildTarget,
                              objects: T.Sequence[T.Union[str, 'File', build.ExtractedObjects]],
-                             proj_dir_to_build_root: str) -> T.Tuple[T.List[str], T.List[build.BuildTargetTypes]]:
-        obj_list: T.List[str] = []
-        deps: T.List[build.BuildTargetTypes] = []
+                             proj_dir_to_build_root: str,
+                             obj_list: T.List[str], deps: T.List[build.BuildTargetTypes]) -> None:
```

Comment:
@dcbaker I suspect you may have some opinions about this.

Reply:
--verbose? :)